nucypher/policy/conditions/address.py

```diff
@@ -0,0 +1,144 @@
+    """
+
+    CONDITION_TYPE = ConditionType.ADDRESS_ALLOWLIST.value
+
+    class Schema(AccessControlCondition.Schema):
+        condition_type = fields.Constant(ConditionType.ADDRESS_ALLOWLIST.value)
+        addresses = fields.List(
+            fields.String(required=True),
+            required=True,
+            validate=validate.Length(min=1, max=25),
+        )
+
+    def __init__(
+        self,
+        addresses: List[str],
+        name: Optional[str] = None,
+    ):
+        """
+        Initialize a AddressAllowlistCondition.
+
+        Args:
+            addresses: List of checksummed Ethereum addresses that are allowed to decrypt
+            name: Optional name for the condition
+        """
+        # Check for duplicates
+        if len(set(addresses)) != len(addresses):
+            raise InvalidCondition("Duplicate addresses are not allowed")
+
+        # Validate all addresses
+        for address in addresses:
+            try:
+                # Ensure addresses have proper checksum
+                normalized_address = to_checksum_address(address)
+            except ValueError as e:
+                raise InvalidCondition(f"Invalid Ethereum address: {address}") from e
+            if normalized_address != address:
+                raise InvalidCondition(
+                    f"Address {address} is not a checksummed address"
+                )
+
+        # Store the validated addresses
+        self.addresses = addresses
+
+        super().__init__(
+            condition_type=self.CONDITION_TYPE,
+            name=name,
+        )
+
+    def verify(self, **context) -> Tuple[bool, Union[None, Dict]]:
+        """
+        Verify if the user's wallet address is in the allowlist.
+
+        Args:
+            context: Dictionary containing context data, including the user's address
+
+        Returns:
+            Tuple containing:
+                - Boolean indicating if verification passed
+                - None (no additional data returned)
+        """
+        if not context:
+            raise RequiredContextVariable("No value provided for context variable")
+
+        # Get user's address using resolve_any_context_variables
+        user_address = resolve_any_context_variables(context[USER_ADDRESS_CONTEXT])[
+            "address"
+        ]
+        # Simply check if the normalized address is in the allowlist
+        is_allowed = to_checksum_address(user_address) in self.addresses
+
+        return is_allowed, None
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, AddressAllowlistCondition)
+            and self.condition_type == other.condition_type
+            and set(self.addresses) == set(other.addresses)
+            and self.name == other.name
+        )
+
+    def __repr__(self):
+        addresses_str = ", ".join(self.addresses[:3])
+        if len(self.addresses) > 3:
+            addresses_str += f"... (+{len(self.addresses) - 3} more)"
```

Comment:
I would just print the length of addresses, rather than printing any specific addresses. Also helps with privacy concerns of logs.